integration/test_collection.py

```diff
@@ -785,15 +781,9 @@ class BPropsInsert(TypedDict):
             ],
             vectorizer=Vectorizer.NONE,
         ),
-        BPropsInsert,
-    )
-    uuid_B = B.data.insert(BPropsInsert(name="B", ref=ReferenceTo(uuids=uuid_A1)))
-    B.data.reference_add(from_uuid=uuid_B, from_property="ref", ref=ReferenceTo(uuids=uuid_A2))
-
-    class CPropsInsert(TypedDict):
-        name: str
-        ref: ReferenceTo
-        age: int
+    ).data.with_data_model(BProps)
+    uuid_B = B.insert(properties=BProps(name="B", ref=Reference[AProps].to(uuids=uuid_A1)))
```

Comment:
I think it will be once mypy runs on the test, which ofc isn't now!

Reply:
hmmm but we need an UUID here and nothing else - eg the type doesn't make a difference and so users shouldn't need to provide it if possible

Reply:
> To be clear, the confusing type here is the generic AProps and Reference.to(uuids=uuids_A1) would be preferable?

yes

> If a user is making use of static type compile-time checks they will be gladly bound by the type checker. This is an example where it would be useful and helpful to them to ensure that their BProps and AProps are defined appropriately to one another (It's also an outcome of coupling the insert and output types)

When creating the class: Sure, but on inserts I don't see the benefits. There is nothing on the UUID that tells you "this is for A" and the type-checker won't help you to avoid bugs here. Also I don't think we will use the type for anything as it doesn't matter (not sure if mypy is smart enough to notice that)

> Since it is defined that BProps.ref: Reference[AProps] so that BProps can be used in return_properties of the query, they have to provide the BProps object in .insert matching this definition since that is reason for using a static type checker. If they aren't using a type checker then they can easily drop the [AProps] specification and Python won't complain

Is there a way to not have it there even for the typed case without mypy complaining? :)

Reply:
Interestingly, `mypy` doesn't complain about those lines but this could be because all of the other errors we've yet to fix obscuring it. However, since `Reference` is defined generically using the `P` generic, which has no bounds or defaults, this would make sense

The fundamental requirement for the generic in the type definition, e.g. `BProps.ref: Reference[AProps]` is so that the returned dictionary from the query can be typed appropriately but it does indeed have no effect upon insert and this goes for the `mypy` compile checks also

With this in mind, does it need changing? I can alter this test to not specify the generic to make it clearer